### PR TITLE
add 3d input and output shapes to maxpool documentation

### PR DESCRIPTION
**Pull request by migeed-z**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61790
* #61776
* #61747
* **#61310**

Differential Revision: [D29737516](https://our.internmc.facebook.com/intern/diff/D29737516)

### CODE REVIEW

File: torch/nn/modules/pooling.py [hunk 1]
```diff
@@ -95,7 +95,8 @@ class MaxPool2d(_MaxPoolNd):
     planes.
 
     In the simplest case, the output value of the layer with input size :math:`(N, C, H, W)`,
-    output :math:`(N, C, H_{out}, W_{out})` and :attr:`kernel_size` :math:`(kH, kW)`
+    output :math:`(N, C, H_{out}, W_{out})` or with input size :math:`(C, H, W)`,
```
Comment (jbschlosser) [inline torch/nn/modules/pooling.py:98]: Note that the update here doesn't fit with the math below, as the math references `N_i`. I'd suggest leaving this part as-is.

File: torch/nn/modules/pooling.py [hunk 2]
```diff
@@ -140,6 +144,7 @@ class MaxPool2d(_MaxPoolNd):
               W_{out} = \left\lfloor\frac{W_{in} + 2 * \text{padding[1]} - \text{dilation[1]}
                     \times (\text{kernel\_size[1]} - 1) - 1}{\text{stride[1]}} + 1\right\rfloor
 
+
```
Comment (jbschlosser) [inline torch/nn/modules/pooling.py:147]: nit: any need for the extra newline?

File: torch/nn/modules/pooling.py [hunk 3]
```diff
@@ -130,7 +131,10 @@ class MaxPool2d(_MaxPoolNd):
 
     Shape:
         - Input: :math:`(N, C, H_{in}, W_{in})`
-        - Output: :math:`(N, C, H_{out}, W_{out})`, where
+        - Output: :math:`(N, C, H_{out}, W_{out})` or
```
Comment (jbschlosser) [inline torch/nn/modules/pooling.py:134]: For consistency with other modules, both possible shapes should be on a single Input and Output line (check out [nn.Upsample](https://pytorch.org/docs/stable/generated/torch.nn.Upsample.html) as an example).

**Review by jbschlosser — COMMENTED** [review]
Thanks for the update! I added a couple nitpicky comments below, but we definitely want the 3D case documented :)

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29737516).

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29737516).

File: torch/nn/modules/pooling.py [hunk 4]
```diff
@@ -129,8 +129,8 @@ class MaxPool2d(_MaxPoolNd):
         ceil_mode: when True, will use `ceil` instead of `floor` to compute the output shape
 
     Shape:
-        - Input: :math:`(N, C, H_{in}, W_{in})`
-        - Output: :math:`(N, C, H_{out}, W_{out})`, where
+        - Input: :math:`(N, C, H_{in}, W_{in})` or :math:`(C, H_{in}, W_{in})`
+        - Output: :math:`(N, C, H_{out}, W_{out})` or :math:`(C, H_{out}, W_{out})` where,
```
Comment (jbschlosser) [inline torch/nn/modules/pooling.py:133]: nit: comma should still be before the where

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29737516).

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29737516).

**Review by jbschlosser — APPROVED** [review]
LGTM!

**Comment by migeed-z** [comment]
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29737516).

**Pull request closed without merging**